cairo/ethereum/cancun/fork.cairo

```diff
@@ -751,26 +878,35 @@ func check_transaction{
     bitwise_ptr: BitwiseBuiltin*,
     keccak_ptr: felt*,
     poseidon_ptr: PoseidonBuiltin*,
-    state: State,
-}(
-    tx: Transaction,
-    gas_available: Uint,
-    chain_id: U64,
-    base_fee_per_gas: Uint,
-    excess_blob_gas: U64,
-) -> TupleAddressUintTupleVersionedHash {
+    block_env: BlockEnvironment,
+}(block_output: BlockOutput, tx: Transaction) -> TupleAddressUintTupleVersionedHashU64 {
     alloc_locals;
+
+    let gas_available = block_env.value.block_gas_limit.value -
+        block_output.value.block_gas_used.value;
+    let blob_gas_available = MAX_BLOB_GAS_PER_BLOCK - block_env.value.excess_blob_gas.value;
```

Comment:
Should raise when there are no more blob gas available, here it can overflow?

Reply:
done